api/session.ts

```diff
@@ -0,0 +1,102 @@
+import { base } from "api/base"
+import { baseWithAuth } from "./base"
+import { UserType } from "api/types"
+import { errorHandler } from "api/utils"
+
+export interface LoginProps {
+  username: string
+  password: string
+}
+
+export interface RegisterProps {
+  username: string
+  password: string
+}
+
+export const login = async ({ username, password }: LoginProps) => {
+  return await base
+    .request({
+      url: "/login",
+      method: "POST",
+      data: {
+        user: {
+          username: username,
+          password: password,
+        },
+      },
+    })
+    .then((response) => {
+      const user = {
+        id: response.data.id,
+        username: response.data.username,
+      }
+      const token = response.headers.authorization
+
+      localStorage.setItem("userToken", JSON.stringify(token))
+      return user
+    })
+    .catch((err) => {
+      const error = {
+        password: err.response.data, // formik only accepts username || password as error
```

Comment:
Doesn't formik support custom errors? Can you explain a bit further. Will accept if it makes sense lol

Reply:
okay, if I remember correctly, for the login -- devise only returns a single data object
![image](https://user-images.githubusercontent.com/91857581/191987755-f051a6de-2915-4360-8af4-2c1d752dbd66.png)
which makes sense since its for the login

eh Formik is looking for an error object that has seperate errors for `username` and `password` thats why I made an error object with the password key so the error gets displayed under the `password input` on the form. 

its not a smooth solution but being unfamiliar with Formik, thats what I was able to think of 😅